format/cache.go

```diff
@@ -0,0 +1,93 @@
+package format
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/charmbracelet/log"
+	"github.com/numtide/treefmt/walk/cache"
+	bolt "go.etcd.io/bbolt"
+)
+
+func CompareFormatters(db *bolt.DB, formatters map[string]*Formatter) error {
+	return db.Update(func(tx *bolt.Tx) error {
+		clearPaths := false
+
+		pathsBucket, err := cache.BucketPaths(tx)
+		if err != nil {
+			return fmt.Errorf("failed to get paths bucket from cache: %w", err)
+		}
+
+		formattersBucket, err := cache.BucketFormatters(tx)
+		if err != nil {
+			return fmt.Errorf("failed to get formatters bucket from cache: %w", err)
+		}
+
+		// check for any newly configured or modified formatters
+		for name, formatter := range formatters {
+
+			stat, err := os.Lstat(formatter.Executable())
```

Comment:
On Windows, the Lstat would fail because of the extra `.exe` file extension executables have. Nothing to fix immediately, but it's one of those places that would have to be tweaked for Windows support.

Reply:
Good to know. I think that's something we would tackle in a specific PR targeting windows support.